consistency-tests/__tests__/clickables.test.js

```diff
@@ -156,3 +163,133 @@ describe.each`
         expect(clickHandler).toHaveBeenCalled();
     });
 });
+
+/* ******* tabIndex tests ******* */
+
+// These components should all wrap buttons or links, so they should
+// inherently be clickable and keyboard navigable. They should not have
+// a default tabIndex 0 as that would be redundant.
+describe.each`
+    Component            | name
+    ${ActionItem}        | ${"ActionItem"}
+    ${Button}            | ${"Button"}
+    ${ClickableWrapper}  | ${"Clickable"}
+    ${CompactCell}       | ${"CompactCell"}
+    ${DetailCell}        | ${"DetailCell"}
+    ${IconButtonWrapper} | ${"IconButton"}
+    ${Link}              | ${"Link"}
+`("$name", ({Component, name}) => {
+    test("doesn't have a redundant tabIndex of 0", () => {
+        // Arrange
+
+        // Act
+        render(
+            <Component onClick={jest.fn()} testId="clickable-component-test-id">
+                Click me
+            </Component>,
+        );
+
+        // Assert
+        const component = screen.getByTestId("clickable-component-test-id");
+        expect(component).not.toHaveAttribute("tabIndex");
+    });
+});
+
+// These components do not wrap an inherently clickable component, so
+// it is expected that they have a tabIndex of 0 so they can be keybaord
+// navigable.
+describe.each`
+    Component     | name
+    ${OptionItem} | ${"OptionItem"}
+`("$name", ({Component, name}) => {
```

Comment:
**suggestion (non-blocking):** This could be combined with the previous test by introducing an additional column `hasTabIndex` with boolean values.  This variable could then be used to decide what assertion to use in the test case.